project/user_provision.py

```diff
@@ -74,40 +76,55 @@ def main():
     pluginInstruction = []
     if args.plugin is not None:
         for email in emails:
-            #    x = \
-            runPlugins(configMap, plugins, email, allPermissions, pluginInstruction, availablePlugins, args.name,
-                       arg='add')
-            # changed here
-            #     if x != None:
-            print('sending email')
-            mail.emailOutput(email, configMap, pluginInstruction, arg='add')
+
+            runPlugins(configMap, plugins, email, allPermissions, pluginInstruction, availablePlugins, args.name, arg='add')
+
+            if(registered == True):
+
+                print('\nsending email')
+                mail.emailOutput(email, configMap, pluginInstruction, arg='add')
 
     if args.remove is not None:
+
         for email in emails:
-            runPlugins(configMap, pluginsremove, email, allPermissions, pluginInstruction, availablePlugins, args.name,
-                       arg='remove')
-            print('sending email')
-            mail.emailOutput(email, configMap, pluginInstruction, arg='remove')
+
+            runPlugins(configMap, pluginsremove, email, allPermissions, pluginInstruction, availablePlugins, args.name, arg='remove')
+
+            if (registered == True):
+
+                print('sending email')
+                mail.emailOutput(email, configMap, pluginInstruction, arg='remove')
 
 
 def runPlugins(configMap, plugins, email, allPermissions, pluginInstruction, availablePlugins, name, arg):
+    global registered
```

Comment:
Rather than using a global, why not have runPlugins return a boolean just indicating if there was an error in any plugin?  Or, runPLugins could return a struct with the status of each plugin....may be helpful later?